api/recover.go

```diff
@@ -48,11 +42,8 @@ func (a *API) Recover(w http.ResponseWriter, r *http.Request) error {
 		referrer := a.getReferrer(r)
 		return a.sendPasswordRecovery(tx, user, mailer, config.SMTP.MaxFrequency, referrer)
 	})
-	if err != nil {
-		if errors.Is(err, MaxFrequencyLimitError) {
-			return tooManyRequestsError("For security purposes, you can only request this once every 60 seconds")
-		}
-		return internalServerError("Error recovering user").WithInternalError(err)
```

Comment:
same here, this error could possibly be due to an error from the transaction above, we need to check it and probably return a more generic error message than not at all